### PR TITLE
Bugfixes, refactoring, and updated docs for r.sim.terrain

### PR DESCRIPTION
**Pull request by baharmon**
**Changes**
- Fixed a bug for running 'series' mode
- Edge effect handling switched from r.grow.distance to r.slope.aspect's '-e' flag
- Switched to grass script's 'mapcalc' wrapper
- Switched from string formatting to f-strings where appropriate
- Applied Black formatting
- Updated troubleshooting in manual page
- Updated references in manual page

**Review**
Code review requested for Python style @wenzeslaus

### CODE REVIEW

File: grass7/raster/r.sim.terrain/r.sim.terrain.html [hunk 1]
```diff
@@ -93,19 +93,25 @@ <h2>EXAMPLES</h2>
...
-
 then a lower <b>nwalkers</b> parameter value has to be selected.
 
+If the module fails with
+<div class="code"><pre>
+ERROR: Unable to insert dataset of type raster in the temporal database. The mapset of the dataset does not match the current mapset.
+</pre></div>
+check that the input  <b>elevation</b> map is in the current mapset.
+The input <b>elevation</b> map must be in the current mapset
+so that it can be registered in the temporal datasbase.
```
Comment (petrasovaa) [inline grass7/raster/r.sim.terrain/r.sim.terrain.html:107]: typo datasbase

File: grass7/raster/r.sim.terrain/r.sim.terrain.html [hunk 2]
```diff
@@ -93,19 +93,25 @@ <h2>EXAMPLES</h2>
 <h2>ERROR MESSAGES</h2>
 
 If the module fails with
-
 <div class="code"><pre>
 ERROR: nwalk (7000001) &gt; maxw (7000000)!
 </pre></div>
-
```
Comment (petrasovaa) [inline grass7/raster/r.sim.terrain/r.sim.terrain.html:100]: This error should not show up anymore with 7.8.5 I believe, the restriction on number of cells is removed. The advice would not work anyway. I suggest removing this from here.

File: grass7/raster/r.sim.terrain/r.sim.terrain.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
-
+# -*- coding: utf-8 -*-
```
Comment (petrasovaa) [inline grass7/raster/r.sim.terrain/r.sim.terrain.py:2]: not needed for Python3, please remove

File: grass7/raster/r.sim.terrain/r.sim.terrain.py [hunk 4]
```diff
@@ -2312,22 +1973,17 @@ def rainfall_series(self):
...
-                # remove temporary maps
-                gscript.run_command(
-                    "g.remove", type="raster", name=["rain_excess"], flags="f"
-                )
+                try:
+                    # remove temporary maps
+                    gscript.run_command(
+                        "g.remove", type="raster", name=["rain_excess"], flags="f"
+                    )
+                except (NameError, CalledModuleError):
```
Comment (petrasovaa) [inline grass7/raster/r.sim.terrain/r.sim.terrain.py:1981]: Not sure why NameError here

**Review by petrasovaa — COMMENTED** [review]
With flake8 I am getting:
```
/tmp/r.sim.terrain.py:383:1: F401 'os' imported but unused
/tmp/r.sim.terrain.py:388:1: F401 'math.exp' imported but unused
/tmp/r.sim.terrain.py:699:9: F841 local variable 'aspect' is assigned to but never used
/tmp/r.sim.terrain.py:911:9: F841 local variable 'sedflux' is assigned to but never used
/tmp/r.sim.terrain.py:1497:22: F821 undefined name 'depth'
```

**Comment by wenzeslaus** [comment]
You just happen to open this PR during transition from using the master branch to using a separate branch for each major version. Please, change the base branch for this PR from master to grass7 using the _Edit_ button next to the PR title.

This will create conflict. Although, the web interface doesn't allow you to merge the base branch into your branch in this case, the command line git tool should handle it without any additional steps or settings (at least version 2.25.1). Standard fetch + rebase should do the trick:

```
git switch my-feature-branch
git fetch upstream
git rebase upstream/grass7
```

Git identifies the move and applies your changes to the moved files. I think `git merge upstream/grass7` should do the same, but I'm not sure whether I have ever used it in this case.

In any case, this is easy to try in a new branch:

```
git switch my-feature-branch
git switch -c test-src-rename-my-feature-branch
git fetch upstream
git rebase upstream/grass7
```

Then use `git status`, `git log`, and `ls` to investigate what happened. If it looks okay, do it on the original feature branch. (After the PR is merged, you can delete both branches.)

If you do rebase, you will need to do force push (`git push -f`) instead of push because the commit hashes will be different (which is true anytime you do rebase). Before force push, it is good to confirm you are on the right branch using `git status`. If you do merge, there is no need for force push, just push.

FYI, this is the same workflow as any other time the master branch contains work you want in your branch.

**Comment by baharmon** [comment]
Replaced by #578.

**Pull request closed without merging**